modules/upcloud.py

```diff
@@ -278,7 +278,7 @@ def main():
             core_number = dict(type='int'),
             memory_amount = dict(type='int'),
             ip_addresses = dict(type='list'),
-            firewall = dict(type='bool'),
+            firewall = dict(type='str'),
```

Comment:
Maybe choises would work here, like in the state definition.
state = dict(choices=['present', 'absent'], default='present'),